app/views/users/index.html.erb

```diff
@@ -1,246 +1,18 @@
-
+<%=console if Rails.env == "development"%>
 <section class="users-section">
   <ul class="users-list">
     <% @users.each do |user| %>
       <li>
         Name: <%= user.name %>
         <span class="profile-link">
           <%= link_to 'See Profile',  user_path(user), class: 'profile-link' %>
+          <%if !current_user.friend?(user) &&  current_user != user %>
```

Comment:
**[REQUIRED]**
* I can be able to send a friend request twice since the request button is there even after sending a friend requests.
To fix this, we can define a helper method in `app/helpers` like this:
```ruby
module UsersHelper
    def pending_friend?(user)
        current_user.pending_friends.include?(user) || current_user.friend_requests.include?(user)
    end
end
```
This will check if the `current_user` has already sent the friend request to the target user or if the target user has already sent a friend request to the current_user.
We can now update the views like this:
```ruby
<%if !current_user.friend?(user) &&  current_user != user && !pending_friend?(user) %>
```

Reply:
Thank you for your awesome review with a lot of detail! nothing more to be desired.